features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapReceiverSnmp4jImpl.java

```diff
@@ -42,29 +40,52 @@
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
+import javax.annotation.Resource;
+
 import org.opennms.core.concurrent.LogPreservingThreadFactory;
 import org.opennms.core.logging.Logging;
 import org.opennms.core.utils.InetAddressUtils;
-import org.opennms.netmgt.config.SyslogdConfig;
+import org.opennms.netmgt.config.TrapdConfig;
+import org.opennms.netmgt.snmp.SnmpUtils;
+import org.opennms.netmgt.snmp.SnmpV3User;
+import org.opennms.netmgt.snmp.TrapNotification;
+import org.opennms.netmgt.snmp.TrapNotificationListener;
+import org.opennms.netmgt.snmp.TrapProcessor;
+import org.opennms.netmgt.snmp.TrapProcessorFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 
 /**
  * @author <a href="mailto:weave@oculan.com">Brian Weaver</a>
  * @author <a href="http://www.oculan.com">Oculan Corporation</a>
  * @fiddler joed
  */
-// TODO: HZN-609: Change to implement TrapNotificationListener
-public class TrapReceiverSnmp4jImpl implements TrapReceiver {
+public class TrapReceiverSnmp4jImpl implements TrapReceiver, TrapNotificationListener,TrapProcessorFactory {
     private static final Logger LOG = LoggerFactory.getLogger(TrapReceiverSnmp4jImpl.class);
 
-    private static final int SOCKET_TIMEOUT = 500;
+    /**
+     * The thread pool that processes traps
+     */
+    private ExecutorService m_backlogQ;
 
```

Comment:
Get rid of m_backlogQ. Instead of using it, we will be sending messages to the list of handlers inside m_trapNotificationHandlers.